### PR TITLE
WIP Load PR template

### LINKED ISSUE #5223: Pull text from github repo when creating a pull request in application
**Do you want to request a *feature* or report a *bug*?**
FEATURE
**What is the current behavior?**
Pull text from the new pull request page of the target repo.
image

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
The body to contain the template, if any, of the new issue.  The best way is to pull down https://raw.githubusercontent.com/{dynamic repo value}/master/.github/PULL_REQUEST_TEMPLATE.md
**Environment you encounter the issue:**

- GitExtensions version:Master
- GIT version: NA
- OS version: NA
- .NET version: NA

**Did this work in previous version of GitExtensions (which)?**

### PR DESCRIPTION
**Pull request by Yiabiten**
Fixes #5223

Changes proposed in this pull request:
- [x] Load PR template from github
- [ ] make it async
- [ ] Reload when branch is changed

 
Screenshots before and after (if PR changes UI):
- N/A

What did I do to test the code and ensure quality:
- Run it

Has been tested on (remove any that don't apply):
- GIT 2.11 and above
- Windows 7 and above

### CODE REVIEW

**Comment by drewnoakes** [comment]
Great to see progress on this.

Why did you decide to source the template via HTTP rather than from the local `.github` folder?

**Comment by Yiabiten** [comment]
To have the same behavior as the github website's "New Pull Request" Click: Always get the latest pushed template (not the local -possibily edited- version)
Getting the local copy is easier and less painfull to do 😄. Let's see what the others prefer before I continue..

**Comment by drewnoakes** [comment]
These templates don't change too often. I would say it'd be a better experience overall to try and source the file locally. Happy with either approach if others disagree.

**Comment by drewnoakes** [comment]
@Yiabiten will you have a chance to continue work on this?

**Comment by vbjay** [comment]
The problem with sourcing locally is if the template was changed and they are on a branch that is before the template change, they won't get the updated template.

**Comment by vbjay** [comment]
FYI.  #5955

File: GitUI/CommandsDialogs/RepoHosting/CreatePullRequestForm.cs [hunk 1]
```diff
@@ -112,6 +115,26 @@ private void _pullReqTargetsCB_SelectedIndexChanged(object sender, EventArgs e)
         [CanBeNull]
         private IHostedRemote MyRemote => _hostedRemotes.FirstOrDefault(r => r.IsOwnedByMe);
 
+        private void LoadPRTemplate()
+        {
```
Comment (RussKie) [inline GitUI/CommandsDialogs/RepoHosting/CreatePullRequestForm.cs]: There is a room for improvement - the operation must not be blocking, all IO must happen on a background thread. 
Please have a look how we download avatars:
https://github.com/gitextensions/gitextensions/blob/a177c46a2f0e7a42481d21e2f5638d9d610e1b95/GitUI/Avatars/AvatarDownloader.cs#L153-L187
I think for the most part you can reuse the code (you won't need semaphores).

Have you thought of failure scenarios? E.g. 
* network is unavailable
* timeouts
* HTTP 404
* HTTP 500, etc

What is a proposal for UX?

**Comment by GintasS** [comment]
This is an old PR and should be closed or set to a draft. @RussKie

**Pull request closed without merging**